File: backend/src/main/java/com/yangshengzhou/lucky_sms/pojo/TeachingAssignment.java
package com.yangshengzhou.lucky_sms.pojo;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 教学安排实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("teaching_assignments")
public class TeachingAssignment extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Integer assignmentId;
    
    @TableField("course_id")
    private Integer courseId;
    
    @TableField("teacher_id")
    private Integer teacherId;
    
    @TableField("semester_id")
    private Integer semesterId;
    
    @TableField("class_time")
    private String classTime;
    
    private String classroom;
    
    @TableField("max_students")
    private Integer maxStudents;
    
    @TableField("current_students")
    private Integer currentStudents;
    
    @TableField("credit_hours")
    private Integer creditHours;
    
    @TableField("course_type_id")
    private Integer courseTypeId;
    
    @TableField("assessment_method_id")
    private Integer assessmentMethodId;
    
    @TableField("status_id")
    private Integer statusId;
    
    private String description;
    private String requirements;
    

    @TableField(exist = false)
    private Course course;
    
    @TableField(exist = false)
    private Teacher teacher;
    
    @TableField(exist = false)
    private Semester semester;
    
    @TableField(exist = false)
    private CourseType courseType;
    
    @TableField(exist = false)
    private AssessmentMethod assessmentMethod;
    
    @TableField(exist = false)
    private StatusType status;
}

File: backend-go/models/teaching_assignment.go
package models

type TeachingAssignment struct {
	BaseEntity
	AssignmentID int `json:"assignmentId" gorm:"column:assignment_id;primaryKey;autoIncrement"`
	TeacherID    int `json:"teacherId" gorm:"column:teacher_id;index"`
	CourseID     int `json:"courseId" gorm:"column:course_id;index"`
	SemesterID   int `json:"semesterId" gorm:"column:semester_id;index"`
	StatusID     int `json:"statusId" gorm:"column:status_id"`

	Teacher  *Teacher  `json:"teacher,omitempty" gorm:"-"`
	Course   *Course   `json:"course,omitempty" gorm:"-"`
	Semester *Semester `json:"semester,omitempty" gorm:"-"`
}

func (TeachingAssignment) TableName() string {
	return "teaching_assignments"
}
